src/App.tsx

```diff
@@ -1,24 +1,260 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useState, useCallback, useEffect, useMemo,
+} from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { Todo, PathchingTodo } from './types/Todo';
+import {
+  getTodos, postTodo, deleteTodo, patchTodo, USER_ID,
+} from './api/todos';
+import TodoList from './components/TodoList';
+import { FilterType } from './types/FilterType';
+import Footer from './components/Footer';
+import HeaderForm from './components/HeaderForm';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [disabledInput, setDisabledInput] = useState(false);
+  const [query, setQuery] = useState('');
+  const [errorText, setErrorText] = useState('');
```

Comment:
still didn't fixed https://github.com/mate-academy/react_todo-app-with-api/pull/413/files#r1169896011

Reply:
It won't be more complicated. Split-to-components code looks more clear. Less code in one file is better)
Also using ErrorMessage will make your code more predictable